### DISCUSSION TITLE
Is there a way to help Frigate not getting crazy in case of fog  ?

### DISCUSSION THREAD

**Discussion by bonjour81** [blakeblackshear/frigate#15547] [Ask A Question]
Hi,
Last night was foggy and I noticed my coral gets hot (56°C) with a pretty higher load than usual (>20% vs usually ~1%).
It’s simply frigate detecting mouvement everywhere, all the time due to this IR interaction with fog (see screenshot,  imagine boxes jumping everywhere)

![Screenshot_20241217_064144_Firefox](https://github.com/user-attachments/assets/c5526fd6-8947-4abf-bcc6-3d707ea6dc15)

Is there anyway to improve this, even partly, in Frigate or in camera ?  
I mean except using external IR illuminator and Without downgrading too much detection in low light.
Does external illuminator solve it efficiently or not always ?

**Comment by hh-TDA** [comment]
Are you tuning with Motion Tuner?

**Comment by NickM-27** [comment]
Fog is a difficult scenario for motion detection and something we hope to improve in the future. You could setup an automation to detect this high detect fps and disable or temporarily tune motion detection

**Comment by abrahamneben** [comment]
I know we can set a speed threshold to exclude slow moving objects, but would it help here to set a _maximum_ speed threshold to exclude these spurious fast moving detections? 

![Screen Recording 2026-03-07 at 10 39 49 PM](https://github.com/user-attachments/assets/381ad8ac-f341-424b-9e78-077b74488abf)

**Comment by tinimark** [comment]
I was getting the same problems last night (very foggy). 
Recently changed to Yolov9 and false positives have been much better but the fog really set things going crazy seeing cars and people that were not there triggering automations etc.
Annoyingly, all these false positives didn't show up with snapshots I can't send them to Frigate+ to try and tune them out.
